### PR TITLE
VSTS build and deploy

### PR DESCRIPTION
**Pull request by chelnak**
### Context
This PR adds support for azure piplines build and releases

### Changes proposed in this pull request
* Add ARM templates to deploy infrastructure
* Add azure-pipelines.yml
* add rspec_junit_formatter gem to enable publishing test results
* Allow multi-file docker-compose builds in az pipelines

### Guidance to review
Does everything look OK?

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+pr: none
+
+pool:
+  vmImage: 'Ubuntu-16.04'
+
+variables:
+  imageName: 'manage-courses-support-poc'
+  # Support multiple docker-compose files to allow travis and az pipelines to co exist: https://docs.docker.com/compose/extends/#multiple-compose-files
+  dockerOverride: 'docker-compose -f docker-compose.yaml -f docker-compose.az.yml'
```
Comment (misaka) [inline azure-pipelines.yml]: Just curious ... below I see `$DOCKER_OVERRIDE` being used, which I would have to guess is set by this setting above, but then there's also `$(imageName)` used, which I would've guessed could've been `$IMAGE_NAME`, based on the setting in this section. Is that the case? Or has `DOCKER_OVERRIDE` been set by something else?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    echo "##vso[build.updatebuildnumber]$GIT_SHORT_SHA"
+    echo "##vso[task.setvariable variable=docker_path;]$docker_path"
+  displayName: 'Set version number'
+
+- script: |
+    $DOCKER_OVERRIDE build
+    $DOCKER_OVERRIDE up --no-build -d
+    $DOCKER_OVERRIDE exec -T web /bin/sh -c "./wait-for-it.sh db:5432"
+    $DOCKER_OVERRIDE exec -T web /bin/sh -c "bundle exec rails db:setup"
+    $DOCKER_OVERRIDE exec -T web /bin/sh -c "bundle exec rake assets:precompile"
```
Comment (misaka) [inline azure-pipelines.yml]: Does `assets:precompile` require the db? If not, how about we do it before the `wait-for-it.sh` to make better use of time?

File: azure-pipelines.yml (hunk 1 above)
Comment (chelnak) [reply azure-pipelines.yml]: Good point and I will test this today to see if we can clean up the build a tiny bit

File: azure-pipelines.yml (hunk 2 above)
Comment (chelnak) [reply azure-pipelines.yml]: Lets check with @benilovj to make sure it doesn't break anything.

**Pull request merged**